modules/luci-mod-status/htdocs/luci-static/resources/view/status/include/60_wifi.js

```diff
@@ -2,90 +2,45 @@
 'require baseclass';
 'require dom';
 'require network';
+'require uci';
+'require fs';
 'require rpc';
 
-var callSessionAccess = rpc.declare({
-	object: 'session',
-	method: 'access',
-	params: [ 'scope', 'object', 'function' ],
-	expect: { 'access': false }
-});
-
-function renderbox(radio, networks) {
-	var chan = null,
-	    freq = null,
-	    rate = null,
-	    badges = [];
-
-	for (var i = 0; i < networks.length; i++) {
-		var net = networks[i],
-		    is_assoc = (net.getBSSID() != '00:00:00:00:00:00' && net.getChannel() && !net.isDisabled()),
-		    quality = net.getSignalPercent();
-
-		var icon;
-		if (net.isDisabled())
-			icon = L.resource('icons/signal-none.png');
-		else if (quality <= 0)
-			icon = L.resource('icons/signal-0.png');
-		else if (quality < 25)
-			icon = L.resource('icons/signal-0-25.png');
-		else if (quality < 50)
-			icon = L.resource('icons/signal-25-50.png');
-		else if (quality < 75)
-			icon = L.resource('icons/signal-50-75.png');
-		else
-			icon = L.resource('icons/signal-75-100.png');
-
-		var badge = renderBadge(
-			icon,
-			'%s: %d dBm / %s: %d%%'.format(_('Signal'), net.getSignal(), _('Quality'), quality),
-			_('SSID'), net.getActiveSSID() || '?',
-			_('Mode'), net.getActiveMode(),
-			_('BSSID'), is_assoc ? (net.getActiveBSSID() || '-') : null,
-			_('Encryption'), is_assoc ? net.getActiveEncryption() : null,
-			_('Associations'), is_assoc ? (net.assoclist.length || '-') : null,
-			null, is_assoc ? null : E('em', net.isDisabled() ? _('Wireless is disabled') : _('Wireless is not associated')));
-
-		badges.push(badge);
-
-		chan = (chan != null) ? chan : net.getChannel();
-		freq = (freq != null) ? freq : net.getFrequency();
-		rate = (rate != null) ? rate : net.getBitRate();
-	}
-
-	return E('div', { class: 'ifacebox' }, [
-		E('div', { class: 'ifacebox-head center ' + (radio.isUp() ? 'active' : '') },
-			E('strong', radio.getName())),
-		E('div', { class: 'ifacebox-body left' }, [
-			L.itemlist(E('span'), [
-				_('Type'), radio.getI18n().replace(/^Generic | Wireless Controller .+$/g, ''),
-				_('Channel'), chan ? '%d (%.3f %s)'.format(chan, freq, _('GHz')) : '-',
-				_('Bitrate'), rate ? '%d %s'.format(rate, _('Mbit/s')) : '-'
-			]),
-			E('div', {}, badges)
-		])
-	]);
-}
-
-function wifirate(rt) {
-	var s = '%.1f\xa0%s, %d\xa0%s'.format(rt.rate / 1000, _('Mbit/s'), rt.mhz, _('MHz')),
-	    ht = rt.ht, vht = rt.vht,
-		mhz = rt.mhz, nss = rt.nss,
-		mcs = rt.mcs, sgi = rt.short_gi;
-
-	if (ht || vht) {
-		if (vht) s += ', VHT-MCS\xa0%d'.format(mcs);
-		if (nss) s += ', VHT-NSS\xa0%d'.format(nss);
-		if (ht)  s += ', MCS\xa0%s'.format(mcs);
-		if (sgi) s += ', ' + _('Short GI').replace(/ /g, '\xa0');
-	}
-
-	return s;
-}
-
 return baseclass.extend({
 	title: _('Wireless'),
 
+	WPSTranslateTbl: {
+		WPS_PBC_STATUS_DISABLE: _('Disabled'),
+		WPS_PBC_STATUS_ACTIVE: _('Active'),
```

Comment:
Probably it'll be easier then I'd just output those shorted strings right away then...

Reply:
I actually like your implementation that prints the symbol... That way we can enum all the different strings and can be translated by the webui... (in short I had to do the same exact thing even if ubus reported the text directly)